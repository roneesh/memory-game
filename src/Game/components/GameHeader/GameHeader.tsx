import * as React from 'react';

interface IGameHeaderProps {
  finished: boolean;
  score: number;
}

class Game extends React.Component<IGameHeaderProps, {}> {
  public render() {
    const { score, finished } = this.props;
    return (
      <div>
        <h1>Memory</h1>
        <p>score: {score}</p>
        <p>{finished ? 'Game over!' : 'Game in progress!'}</p>
      </div>
    );
  }
}

export default Game;
