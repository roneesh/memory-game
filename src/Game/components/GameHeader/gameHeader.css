.game-header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.game-header__heading {
  font-family: 'Monoton', cursive;
  font-size: 72px;
  margin-bottom: 20px;
}

.game-header__information {
  font-family: 'Raleway', sans-serif;
  font-size: 32px;
  margin-bottom: 10px;
  text-transform: uppercase;
}
