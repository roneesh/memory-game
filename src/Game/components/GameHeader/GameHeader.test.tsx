import * as React from 'react';
import renderer from 'react-test-renderer';
import GameHeader from './GameHeader';

describe('The GameHeader', () => {
  it('it renders a title and props', () => {
    const gameHeader = renderer
      .create(<GameHeader score={0} finished={false} />)
      .toJSON();
    expect(gameHeader).toMatchSnapshot();
  });

  it('it renders when the game is over', () => {
    const gameHeader = renderer
      .create(<GameHeader score={0} finished={true} />)
      .toJSON();
    expect(gameHeader).toMatchSnapshot();
  });
});
