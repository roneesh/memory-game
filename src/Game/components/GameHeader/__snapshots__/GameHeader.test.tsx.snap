// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GameHeader it renders a title and props 1`] = `
<div>
  <h1>
    Memory
  </h1>
  <p>
    score: 
    0
  </p>
  <p>
    Game in progress!
  </p>
</div>
`;

exports[`The GameHeader it renders when the game is over 1`] = `
<div>
  <h1>
    Memory
  </h1>
  <p>
    score: 
    0
  </p>
  <p>
    Game over!
  </p>
</div>
`;
